test/rewards.js

```diff
@@ -68,32 +71,33 @@ contract('Rewards', (accounts) => {
       Rewards.link(RewardsStorageProxy, rewardsStorageProxy.address),
       Rewards.link(SafeMath, safeMath.address)
     ])
-
-    // We mint 1000 tokens and transfer half of them to the testing wallet address
-    proofToken = await ProofToken.new()
-    await proofToken.mint(fund, 1000)
-    await proofToken.transfer(wallet, 500, { from: fund })
-
-    // We deploy the rest of the contracts. The Proof tokens are already allocated before the first epoch
-    store = await Store.new()
-    cryptoDollar = await CryptoDollar.new(store.address)
-    rewards = await Rewards.new(store.address, proofToken.address)
-    cryptoFiatHub = await CryptoFiatHub.new(cryptoDollar.address, store.address, proofToken.address, rewards.address)
-
-    await Promise.all([
-      store.authorizeAccess(cryptoFiatHub.address),
-      store.authorizeAccess(cryptoDollar.address),
-      store.authorizeAccess(rewards.address),
-      cryptoDollar.authorizeAccess(cryptoFiatHub.address),
-      cryptoFiatHub.initialize(blocksPerEpoch)
-    ])
-
-    creationBlockNumber = await cryptoFiatStorageProxy.getCreationBlockNumber(store.address)
-    epoch1 = creationBlockNumber.plus(blocksPerEpoch).toNumber()
-    epoch2 = creationBlockNumber.plus(2 * blocksPerEpoch).toNumber()
   })
 
   describe('State variables', async() => {
+    beforeEach(async() => {
```

Comment:
I see that you moved this beforeEach block from the overall suite beforeEach to each individual suite's beforeEach. I cannot detect any differences between each of the ones defined in the individual beforeEach blocks. Is there a specified reason why you would want to define this multiple times redundantly rather than keep it in the overall suite beforeEach block? If there is no difference between the individuals, then keeping it in the overall suite beforeEach would have the same exact effect.